incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -581,6 +588,37 @@ func (r *HierarchyConfigReconciler) writeHierarchy(ctx context.Context, log logr
 	return true, nil
 }
 
+// deleteSingletonIfExists deletes the singleton in the namespace if it exists.
+// Note: Make sure there's no finalizers on the singleton before calling this
+// function.
+func (r *HierarchyConfigReconciler) deleteSingletonIfExists(ctx context.Context, log logr.Logger, nm string) error {
+	inst, deletingCRD, err := r.getSingleton(ctx, nm)
+	if err != nil {
+		return err
+	}
+
+	// Early exit if the singleton doesn't exist.
+	if inst.CreationTimestamp.IsZero() {
+		return nil
+	}
+
+	log.Info("Deleting illegal HierarchyConfiguration in excluded namespace")
+
+	// If the CRD is being deleted, we don't need to delete it separately. It will
+	// be deleted with the CRD.
+	if deletingCRD {
+		return nil
+	}
+
```

Comment:
This is a bit odd. We say we're deleting the HC but we might actually not do anything. Imagine a case where the HC _isn't_ being deleted and the logs would look weird.

How about:

```
if deletingCRD {
  log.Info("Already deleting illegal HC in excluded namespace")
  return nil
}
log.Info("Deleting illegal...")
```

Reply:
sg. The first log, I used `Already deleting illegal HC in excluded namespace since the CRD is being deleted`.

Reply:
Discussed offline, updated to `HC in excluded namespace is already being deleted` and `Deleting illegal HC in excluded namespace`